plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlVectorType.java

```diff
@@ -0,0 +1,226 @@
+                    .hasMessageContaining("infinite value not allowed in vector");
+            assertThatThrownBy(() -> onRemoteDatabase().execute("SELECT v <-> '[+Infinity]' FROM " + table.getName()))
+                    .hasMessageContaining("infinite value not allowed in vector");
+        }
+    }
+
+    @Test
+    void testDotProductCompatibility()
+    {
+        // Verify compatibility between Trino dot_product function and pgvector <#> operator
+        try (TestTable table = new TestTable(onRemoteDatabase(), "test_vector", "(v vector(3))");
+                TestView view = new TestView(onRemoteDatabase(), "test_dot_product", "SELECT v <#> '[4,5,6]' FROM " + table.getName())) {
+            onRemoteDatabase().execute("INSERT INTO " + table.getName() + " VALUES ('[1,2,3]')");
+
+            assertThat(query("SELECT -dot_product(v, ARRAY[4,5,6]) FROM " + table.getName()))
+                    .matches("SELECT * FROM tpch." + view.getName());
+
+            assertThatThrownBy(() -> onRemoteDatabase().execute("SELECT v <#> '[NaN]' FROM " + table.getName()))
+                    .hasMessageContaining("NaN not allowed in vector");
+            assertThatThrownBy(() -> onRemoteDatabase().execute("SELECT v <#> '[-Infinity]' FROM " + table.getName()))
+                    .hasMessageContaining("infinite value not allowed in vector");
+            assertThatThrownBy(() -> onRemoteDatabase().execute("SELECT v <#> '[+Infinity]' FROM " + table.getName()))
+                    .hasMessageContaining("infinite value not allowed in vector");
+        }
+    }
+
+    @Test
+    void testCosineDistanceCompatibility()
+    {
+        // Verify compatibility between Trino cosine_distance function and pgvector <=> operator
+        try (TestTable table = new TestTable(onRemoteDatabase(), "test_vector", "(v vector(3))");
+                TestView view = new TestView(onRemoteDatabase(), "test_cosine_distance", "SELECT v <=> '[4,5,6]' FROM " + table.getName())) {
+            onRemoteDatabase().execute("INSERT INTO " + table.getName() + " VALUES ('[1,2,3]')");
+
+            assertThat(query("SELECT cosine_distance(v, ARRAY[4,5,6]) FROM " + table.getName()))
+                    .matches("SELECT * FROM tpch." + view.getName());
+
+            assertThatThrownBy(() -> onRemoteDatabase().execute("SELECT v <=> '[NaN]' FROM " + table.getName()))
+                    .hasMessageContaining("NaN not allowed in vector");
+            assertThatThrownBy(() -> onRemoteDatabase().execute("SELECT v <=> '[-Infinity]' FROM " + table.getName()))
+                    .hasMessageContaining("infinite value not allowed in vector");
+            assertThatThrownBy(() -> onRemoteDatabase().execute("SELECT v <=> '[+Infinity]' FROM " + table.getName()))
+                    .hasMessageContaining("infinite value not allowed in vector");
+        }
+    }
+
+    private SqlExecutor onRemoteDatabase()
```

Comment:
nit: Can this be a part of `TestingPostgreSqlServer` so that we could re-use them across tests ? No changes requested.